plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/WorkspaceAwareCanLoadFromDescriptionHelper.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2019 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ide.xtext.server;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.n4js.scoping.utils.CanLoadFromDescriptionHelper;
+import org.eclipse.xtext.resource.XtextResourceSet;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * A specialization that will create the resources in the correct project specific resource set.
+ */
+@Singleton
+public class WorkspaceAwareCanLoadFromDescriptionHelper extends CanLoadFromDescriptionHelper {
```

Comment:
The naming scheme "WorkspaceAware..." for LSP-specific adjustments is a bit unfortunate, because 1) it does not explicitly highlight the fact that it is specific to the LSP case and 2) the term "Workspace" is also used in Eclipse a lot.

However, I guess most of the Eclipse cases will soon be obsolete anyway (and will be removed), right? In that case it's just temporary, so not a problem.

Reply:
I'd like to keep it like this. If it turns out, that we are confused by the terminology, we should rename this late on. In the LSP context (read: java package), a Workspace is well defined.